### DISCUSSION TITLE
Opening up develop branch for beta testers!

### DISCUSSION THREAD

**Discussion by Griatch** [evennia/evennia#2640] [Evennia Official]
As of today, I'm inviting everyone willing to come help me help test the Evennia `develop` branch - what will become Evennia 1.0. Yes, you have been able to check out the `develop` branch before this, but now I think it's in a state where it's useful to get more eyes on it! :eye: :eye: :mag: 

While the develop branch is definitely fully functional and usable already, it is still, well, under development. There will be bugs, docs are not quite up-to-date yet etc. So if you want to help testing out the bleeding edge, you need to be ready to [report errors](https://github.com/evennia/evennia/issues/new/choose) as you find them!

Until release there are a few things remaining to do:
- Fix (at least most) [known bugs](https://github.com/evennia/evennia/issues) that have popped over time.
- Fix bugs that are discovered by beta testers  (you).
- Update the new [1.0-dev documentation](https://www.evennia.com/docs/1.0-dev/index.html). This will likely take the longest time. This documentation is what you should refer to henceforth (and give feedback on) (The wiki will go away and the ReadTheDocs is still on the 0.9.5 version). 
- Some various administrative things around versioning and distribution.

# Installation 

> If you already have an 0.9.5 Evennia server running, make sure to stop it with `evennia stop`. Also run `deactivate` to stop your existing virtualenv. Delete the virtualenv folder or rename it so the new one we'll create below does not clash.

- Install [Python 3.11](https://www.python.org/downloads/). If you use Debian-based Linux (or Windows WSL), something like `sudo apt install python3.11 python3.11-dev python3.11-venv gcc` should work. Note - if you want to also revert back to 0.9.5, it may be better to use Python 3.10, since that's supported on both branches. 3.11 is 1.0-dev only)
- `python3.11 -m venv evenv` 
- `source evenv/bin/activate` (unix/mac), `\evenv\Scripts\activate` (windows)
- if you don't already have evennia from before: `git clone https://github.com/evennia/evennia.git`
- `cd` to your `evennia` repo folder. You want to be where you see `CHANGELOG.md` in the same folder.
- `git pull` to get the latest 
- `git fetch upstream develop` to fetch the upstream (core evennia) develop branch. If you get an error, ignore this step.
- `git checkout develop`
- `pip install -e .`
- `pip install -r requirements_extra.txt` (these are optional extra third-party libs that contribs need)
- Test so you have the `evennia` command available. `evennia -v` should show `Evennia 1.0-dev`.

If you don't have an existing game dir or don't want to keep anything from your existing game dir, you can just start a new one 
using the normal (updated) [install instructions](https://www.evennia.com/docs/1.0-dev/Setup/Installation.html). If you want to keep/convert your existing game dir, continue below. 

- Make a backup of your old game dir! If you use version control, make sure to commit your current state.
- `cd` to your existing 0.9.5-based game folder (like `mygame`.)
- If you have changed `mygame/web`, _rename_ the folder to `web_0.9.5`. If you didn't change anything (or don't have 
anything you want to keep), you can _delete_ it entirely.
- Copy (from the evennia repo)  `evennia/evennia/game_template/web` to `mygame/` (e.g. using `cp -Rf` or a file manager). This new `web` folder replaces the old one and has a very different structure.
- `evennia migrate`
- `evennia start`

If you made extensive work in your game dir, you may well find that you need to do some (hopefully minor) 
changes to your code before it will start with Evennia 1.0. Some important points: 

- The `evennia/contrib/` folder changed structure - there are now categorized sub-folders, so you have to update 
your imports.
- Any `web` changes need to be moved back from your backup into the new structure of `web/` manually.
- See the [Evennia 1.0 Changelog](https://www.evennia.com/docs/1.0-dev/Setup/Changelog.html#evennia-1-0). I'll compile a more readable list of them as release draws nearer. 

Thanks a lot for those of you wanting to help living on the bleeding edge!

**Comment by changemewtf** [comment]
On Pop!_OS 21.10 (an Ubuntu variant), I got this error while running `pip install -e .`:

```
ModuleNotFoundError: No module named 'distutils.command.bdist_wininst'
```

I was able to fix by also installing the `python3-distutils-extra` package via aptitude.

**Comment by homeofpoe** [comment]
`git fetch upstream develop` didn't work for me.  Using `git fetch origin develop` did.
